### PR TITLE
various fixes and improvements

### PR DESCRIPTION
**Pull request by jasonWoodman**

### CODE REVIEW

File: charts/k8ez/templates/cronjobs.yaml [hunk 1]
```diff
@@ -35,14 +36,17 @@ spec:
           {{- end }}
           {{ end }}
           serviceAccount: {{ $.Release.Name }}
+          enableServiceLinks: {{ $.Values.enableServiceLinks }}
```
Comment (jacobbeasley) [inline charts/k8ez/templates/cronjobs.yaml:39]: niiiiice

File: charts/k8ez/templates/deployment.yaml [hunk 2]
```diff
@@ -31,13 +34,32 @@ spec:
...
-      {{- end }}
-      {{- end }}
+{{- end }}
+{{- end }}
       serviceAccount: {{ .Release.Name }}
+      enableServiceLinks: {{ .Values.enableServiceLinks }}
+      securityContext:
+        fsGroup: {{ .Values.podSecurityContext.fsGroup }}
+      affinity:
+        podAntiAffinity:
```
Comment (jacobbeasley) [inline charts/k8ez/templates/deployment.yaml:48]: I'm gussing this makes them try to schedule on different nodes where possible?

Comment (jasonWoodman) [reply charts/k8ez/templates/deployment.yaml:48]: yep, it tells it to prefer different nodes for scheduling, but to ignore for actual execution.  Meaning they can still run on the same nodes if they get scheduled that way, but it prefers during scheduling not to if possible.

File: charts/k8ez/values.yaml [hunk 3]
```diff
@@ -74,6 +85,24 @@ configData: {}
 # Set secret data key-value pairs that will be loaded into env variables
 secretData: {}
 
+# List of files to load into configmaps.  Supports glob patterns, all files that match will be put into the configmap
+configFiles: []
+  # - fileGlob: files/config.toml
+  #   configMapName: config.toml
+
+# List of extra annotations to be injected into pod templates
+extraAnnotations: {}
```
Comment (jacobbeasley) [inline charts/k8ez/values.yaml:94]: Should the extra-annotations be nested under cronjobs, statefulsets, jobs, etc?

Comment (jasonWoodman) [reply charts/k8ez/values.yaml:94]: i could go either way on it. I suppose ideally you would have a global set of them in the root of the values, then override them with resource specific annotations if you want.  In the use case I have been working with we have been using it for the vault agent, and all those resources need to get the same secrets so the global way was nice.  Not sure if we want to implement further right now or not

Comment (jacobbeasley) [reply charts/k8ez/values.yaml:94]: We can circle back to it later.

**Pull request merged**